source/asynch_socket.c

```diff
@@ -870,12 +875,26 @@ socket_error_t lwipv4_socket_reject(struct socket *socket)
 socket_error_t lwipv4_socket_set_option(struct socket *socket, const socket_proto_level_t level,
         const socket_option_type_t type, const void *option, const size_t optionSize)
 {
-    (void) socket;
     (void) level;
-    (void) type;
-    (void) option;
     (void) optionSize;
-    return SOCKET_ERROR_UNIMPLEMENTED;
+    socket_error_t err = SOCKET_ERROR_UNIMPLEMENTED;
+    switch (type) {
+        case SOCKET_OPT_NAGGLE: {
+            if (socket->family != SOCKET_STREAM)
+                return SOCKET_ERROR_UNIMPLEMENTED;
+            if(option == NULL) {
+                tcp_nagle_disable((struct tcp_pcb *) socket->impl);
```

Comment:
Ah, I see that you use lwIP's method to disable Nagle. In that case, is it still required to call `tcp_output` manually (above)?